sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NormalizeFloatingNumbers.scala

```diff
@@ -123,7 +123,8 @@ object NormalizeFloatingNumbers extends Rule[LogicalPlan] {
       val fields = expr.dataType.asInstanceOf[StructType].fields.indices.map { i =>
         normalize(GetStructField(expr, i))
       }
-      CreateStruct(fields)
+      val struct = CreateStruct(fields)
+      If(IsNull(expr), Literal(null, struct.dataType), struct)
```

Comment:
I'm just wondering if we need a new Literal here. Maybe, can we simple just put `expr`?
```scala
- val struct = CreateStruct(fields)
- If(IsNull(expr), Literal(null, struct.dataType), struct)
+ If(IsNull(expr), expr, CreateStruct(fields))
```

Reply:
The AS-IS also looks good to me, too.

Reply:
literal is better here to simplify the expression tree, and probably has better perf.